simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -50,28 +61,112 @@
+ *       Gauge lastSuccess = Gauge.build()
+ *           .name("my_batch_job_last_success").help("Last time my batch job succeeded, in unixtime.").register(registry);
+ *       lastSuccess.setToCurrentTime();
+ *     } finally {
+ *       durationTimer.setDuration();
+ *       CredentialsProvider creds = new BasicCredentialsProvider();
+ *       creds.setCredentials(
+ *          new AuthScope(new HttpHost("localhost")),
+ *          new UsernamePasswordCredentials("username", "password"));
+ *       CloseableHttpClient httpClient = HttpClients.custom().setDefaultCredentialsProvider(creds).build();
+ *       PushGateway pg = new PushGateway("https", "localhost", "443", httpClient);
+ *       pg.pushAdd(registry, "my_batch_job");
+ *     }
+ *   }
+ * }
+ * </pre>
+ * For information about authentication using the HttpClient see:
+ * <a href="https://hc.apache.org/httpcomponents-client-ga/tutorial/html/authentication.html">
+ * https://hc.apache.org/httpcomponents-client-ga/tutorial/html/authentication.html</a>
+ * <p>
+ * For preemptive-auth see:
+ * <a href="http://www.programcreek.com/java-api-examples/index.php?source_dir=jenkins-client-master/src/main/java/com/offbytwo/jenkins/client/JenkinsHttpClient.java">JenkingsHttpClient</a>
+ * <a href="http://www.programcreek.com/java-api-examples/index.php?source_dir=jenkins-client-master/src/main/java/com/offbytwo/jenkins/client/PreemptiveAuth.java">PreemptiveAuth</a>
+ * <p>
  * See <a href="https://github.com/prometheus/pushgateway">https://github.com/prometheus/pushgateway</a>
  */
 public class PushGateway {
 
-  private final String address;
-  private static final int SECONDS_PER_MILLISECOND = 1000;
+  private static final int TIMEOUT = 10000;
+
+  private final HttpClient httpClient;
+
+  private final String urlBase;
+
+  private BasicHttpContext localContext;
+
+  /**
+   * Construct a Pushgateway, with the given url (without scheme).
+   * The uri will be split by ':' into address and port.
+   * <p>
+   *
+   * @param url host:port or ip:port of the Pushgateway.
+   */
+  public PushGateway(String url) {
+    this(url.split(":")[0], url.split(":")[1]);
+  }
+
+  /**
+   * Construct a Pushgateway, with the given address and port.
+   * <p>
+   *
+   * @param address host or ip of the Pushgateway.
+   * @param port    port of the Pushgateway.
+   */
+  public PushGateway(String address, String port) {
+    this("http", address, port, HttpClients.createDefault());
+  }
+
   /**
    * Construct a Pushgateway, with the given address.
    * <p>
-   * @param address  host:port or ip:port of the Pushgateway.
+   *
+   * @param scheme     address scheme of the PushGateway.
+   * @param address    host or ip of the PushGateway.
+   * @param port       port of the PushGateway.
+   * @param httpClient httpClient for the connection to the PushGateway
+   */
+  public PushGateway(String scheme, String address, String port, HttpClient httpClient) {
```

Comment:
I don't think we need this many consructors. Just one with a URL, and one with the url and client should be sufficient,